### PR TITLE
Set username and password to optional

### PR DESCRIPTION
**Pull request by naitmare01**
This PR will set the `username` and `password` to be optional to be able to run `terraform validate` without providing any credentials i.e. in CI/CD pipelines.

This will fix [issue 125](https://github.com/taiidani/terraform-provider-jenkins/issues/125)

### CODE REVIEW

**Comment by naitmare01** [comment]
@taiidani please review this if you have time. :)

File: jenkins/provider.go [hunk 1]
```diff
@@ -26,13 +26,13 @@ func Provider() *schema.Provider {
...
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true, // Needs to be optional to be able to run terraform validate in CICD pipelines.
 				DefaultFunc: schema.EnvDefaultFunc("JENKINS_USERNAME", nil),
 				Description: "Username to authenticate to Jenkins.",
 			},
 			"password": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true, // Needs to be optional to be able to run terraform validate in CICD pipelines.
```
Comment (taiidani) [inline jenkins/provider.go:35]: I'm not sure we should call out "in CICD pipelines" explicitly here as `terraform validate` may be used in multiple different contexts.

**Review by taiidani — CHANGES_REQUESTED** [review]
Could we also update the provider documentation to reflect these variables no longer being required?

**Comment by naitmare01** [comment]
> Could we also update the provider documentation to reflect these variables no longer being required?

Updated in https://github.com/taiidani/terraform-provider-jenkins/pull/124/commits/74819ce9dd8cfb4c5c991a7de5be221f3eb1c59e

File: jenkins/provider.go (hunk 1 above)
Comment (naitmare01) [reply jenkins/provider.go]: https://github.com/taiidani/terraform-provider-jenkins/pull/124/commits/4e8e7cef683b4294424ee09700469851b176108c

**Pull request merged**